client/connection.go

```diff
@@ -67,6 +67,39 @@ func ConnectLXD(url string, args *ConnectionArgs) (InstanceServer, error) {
 	return httpsLXD(url, args)
 }
 
+// ConnectLXDHTTP lets you connect to a VM agent over a VM socket.
+func ConnectLXDHTTP(vsockID int, args *ConnectionArgs, client *http.Client) (InstanceServer, error) {
```

Comment:
Can you move that part to its own commit so we can later easily revert the agent part of this change to client/ once we have a suitable wrapper in LXD?

Reply:
Done